apps/src/blockly/addons/cdoFieldParameter.ts

```diff
@@ -0,0 +1,265 @@
+    }
+
+    if (!definitionBlock) {
+      const variableName = this.getText();
+      const variable = workspace.getVariable(variableName);
+
+      if (variable) {
+        // Find the first block in the workspace that matches the expected type
+        definitionBlock =
+          workspace.getAllBlocks().find(block => {
+            if (block.type === BLOCK_TYPES.procedureDefinition) {
+              const procedureModel = (
+                block as ProcedureBlock
+              ).getProcedureModel();
+              return procedureModel
+                .getParameters()
+                .some(parameter => parameter.getName() === variableName);
+            }
+            return false;
+          }) || null;
+      }
+    }
+    return {definitionBlock, workspace};
+  }
+  /**
+   * Override of createTextArrow_ to fix the arrow position on Safari.
+   * We need to add dominant-baseline="central" to the arrow element in order to
+   * center it on Safari.
+   *  @override */
+  createTextArrow_() {
+    const arrow = Blockly.utils.dom.createSvgElement(
+      Blockly.utils.Svg.TSPAN,
+      {},
+      this.textElement_
+    );
+    arrow.appendChild(
+      document.createTextNode(
+        this.getSourceBlock()?.RTL
+          ? Blockly.FieldDropdown.ARROW_CHAR + ' '
+          : ' ' + Blockly.FieldDropdown.ARROW_CHAR
+      )
+    );
+
+    /**
+     * Begin CDO customization
+     */
+    arrow.setAttribute('dominant-baseline', 'central');
+    /**
+     * End CDO customization
+     */
+
+    if (this.getSourceBlock()?.RTL) {
+      this.getTextElement().insertBefore(arrow, this.textContent_);
+    } else {
+      this.getTextElement().appendChild(arrow);
+    }
+    // this.arrow is private in the parent.
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (this as any).arrow = arrow;
+  }
+
+  menuGenerator_ = function (this: FieldDropdown): MenuOption[] {
+    // Parameter field dropdowns only have options to rename or delete.
+    return [
+      [commonI18n.renameParameter(), RENAME_PARAMETER_ID],
+      [commonI18n.deleteParameter(), DELETE_PARAMETER_ID],
+    ];
+  };
+
+  /**
+   * Prompt the user for a variable name or delete a variable.
+   * @param promptText description text for window prompt
+   * @param confirmButtonLabel Label of confirm button, e.g. "Rename"
+   * @param callback with parameter (text) of new name
+   * @param prompt Whether to prompt for a string value
+   * @param isDangerCancel Should cancel button has a danger type
+   * @param defaultText default input text for window prompt
+   */
+  static modalDialogName = function (
+    promptText: string,
+    confirmButtonLabel: string,
+    callback: (newName: string) => void,
+    prompt: boolean,
+    isDangerCancel: boolean,
+    defaultText?: string
+  ) {
+    Blockly.customSimpleDialog({
+      bodyText: promptText,
+      prompt,
+      promptPrefill: defaultText,
+      cancelText: confirmButtonLabel,
```

Comment:
why is the cancel text the confirm text?

Reply:
This struck me as odd too. We do it this way for the variable field as well as for the buttons in the modal function editor:
- https://github.com/code-dot-org/code-dot-org/blob/mike/new-param-button/apps/src/blockly/addons/cdoFieldVariable.ts#L133
- https://github.com/code-dot-org/code-dot-org/blob/mike/new-param-button/apps/src/blockly/addons/functionEditor.ts#L425

It didn't seem like I should break from this pattern without some discussion or potential re-working. The rationale seems to be that the dialog has a `isDangerCancel` property that allows us to make the delete buttons red:

https://github.com/code-dot-org/code-dot-org/blob/mike/new-param-button/apps/src/feedback.js#L1290